src/rosdistro/freeze_source.py

```diff
@@ -82,12 +82,30 @@ def freeze_distribution_sources(dist, release_version=False, release_tag=False,
         print("")
 
 
-def _worker(work_queue):
+# Get the repo commit information
+def _get_repo_info(source_repo):
+    count = 0
+    # allow for a retry if the ls-remote fails to query the endpoint repo
     while True:
         try:
-            source_repo, freeze_version, freeze_to_tag = work_queue.get(block=False)
             cmd = ['git', 'ls-remote', source_repo.url]
             ls_remote_lines = subprocess.check_output(cmd).splitlines()
+            return ls_remote_lines
+        except:
+            print("Non-zero return code for: %s retrying command" % ' '.join(cmd), file=sys.stderr)
+            count += 1
+            # reraise the error if we've attempted 3 times
+            if count > 3:
```

Comment:
It would be better if the hard coded retry count and sleep duration were default parameters to the function. They'll be much easier to find there when reading the code.